lib/teach-api.js

```diff
@@ -73,6 +73,20 @@ _.extend(TeachAPI.prototype, {
     }
 
     return req;
+  },
+  getAllData: function(callback) {
```

Comment:
Do you mind renaming this to `getAllClubsData`, since it's very likely that we'll be storing more than just clubs in the Teach API?